pkg/snapshot/snapshotter/snapshotter.go

```diff
@@ -643,6 +644,7 @@ func (ssr *Snapshotter) snapshotEventHandler(stopCh <-chan struct{}) error {
 			if ssr.HealthConfig.SnapshotLeaseRenewalEnabled {
 				ssr.FullSnapshotLeaseUpdateTimer.Stop()
 				ssr.FullSnapshotLeaseUpdateTimer.Reset(time.Nanosecond)
+				FullSnapshotTriggeredTimeCh <- time.Now()
```

Comment:
instead of passing `time.Now()` from snapshotter, why not `RenewFullSnapshotLeasePeriodically()` func will itself calculated this as `RenewFullSnapshotLeasePeriodically` will only start updating the lease when snapshot is taken then why not use this time to calculate `FullSnapshotTriggeredTime`.  
I'm suggesting this to avoid creating more channels for this small improvement and not to make code more complicated.

Reply:
The goroutine `RenewFullSnapshotLeasePeriodically` follows the lifecycle of the snapshotter. Now if the ask is to rely on the call from `case <-ssr.FullSnapshotLeaseUpdateTimer.C`, then it can happen in two cases 1. Fresh full snapshot triggered 2. Previous full snapshot lease update is going on failing, so retry calls also comes to this case itself. So we need a way to distinguish between 1 & 2, that's why I went with having a channel which does this differentiation because the information about the new snapshot has to come from the `snapshotEventHandler`. Do you have any other easy alternative in mind that I can go for?

Reply:
> 1. Fresh full snapshot triggered

yes, when a fresh full snapshot is triggered at that time, you should capture that timestamp to update the in full-snapshot lease.

> 2. Previous full snapshot lease update is going on failing, so retry calls also comes to this case itself. So we need a way to distinguish between 1 & 2

you can easily solve this by check whether timestamp variable is empty or not .
If it's empty then initialize it with current timestamp else don't re-initialize it keep retry with same timestamp.
```go
case <-ssr.FullSnapshotLeaseUpdateTimer.C
// check timestamp is empty or not
// if empty then initialize it with time.Now()
// else don't initialize it, so that it can keep retry with the timestamp it was 

if ssr.PrevFullSnapshot != nil {
  if err := func() error {
	  // retry logic
  } else {
	  //FullSnapshot lease successfully updated. Stop the timer
	  logger.Infof("Stopping the FullSnapshot lease update")
	  ssr.FullSnapshotLeaseUpdateTimer.Stop()
           //  here you can reset the variable to empty, so that it can get re-initialize
  }
  } else {
  //Skip the FullSnapshot lease update as no full snapshot has been taken yet. Reset the timer to retry after interval
  
  }
```

Reply:
I thought of this initially but this won't solve the issue if the retry logic never succeeds till the next full snapshot also, because then the timestamp will not be nil but we should still be resetting the time in this case, so we can't use this logic. 

There is a way, we will have to check the difference in time between currentTime - recorded timestamp and decide. But relying on time duration like that when we actually can get the exact timestamp of when the snapshot has happened with the help of a channel seems not ideal.

Reply:
>  this won't solve the issue if the retry logic never succeeds till the next full snapshot also

this has never happened and seems like a hypothetical case  😕 
but you still want to solve this then why not use this `ssr.PrevFullSnapshot.CreatedOn` timestamp ?

Reply:
> this has never happened and seems like a hypothetical case

yes, since current interval is 24h it may not occur. If full snapshot duration is reduced, then this issue may occur frequently, so I think it's better to cover this 😅 

> to cover this then why not use this ssr.PrevFullSnapshot.CreatedOn timestamp

Oh nice, I didn't realise this field exists that's why I'm doing all this. Then in this case I can just pass this time to `FullSnapshotCaseLeaseUpdate` directly. I'll make this change then